TAO/TAO_IDL/fe/idl.ypp

```diff
@@ -2343,69 +2254,186 @@ positive_int_expr :
         }
         ;
 
-/* Annotations
-   TODO: actions
-   TODO: annotation_dcl and friends
- */
+annotation_dcl
+        : IDL_ANNOTATION_DECL IDENTIFIER '{' annotation_body '}' {
+          if (idl_global->idl_version_ >= IDL_VERSION_4)
```

Comment:
Similar to my note above, should this delegate error/warning output to UTL_Error?  (I'll stop making the same comment now...)